CondTools/SiPhase2Tracker/plugins/SiPhase2BadStripChannelBuilder.cc

```diff
@@ -0,0 +1,295 @@
+// system include files
+#include <vector>
+#include <memory>
+#include <iostream>
+#include <fstream>
+
+// user include files
+#include "CommonTools/ConditionDBWriter/interface/ConditionDBWriter.h"
+#include "CondFormats/SiStripObjects/interface/SiStripBadStrip.h"
+#include "DataFormats/SiStripDetId/interface/StripSubdetector.h"
+#include "DataFormats/TrackerCommon/interface/TrackerTopology.h"
+#include "DataFormats/Phase2TrackerDigi/interface/Phase2TrackerDigi.h"
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+#include "FWCore/ParameterSet/interface/ConfigurationDescriptions.h"
+#include "FWCore/ParameterSet/interface/FileInPath.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/ParameterSet/interface/ParameterSetDescription.h"
+#include "FWCore/ServiceRegistry/interface/Service.h"
+#include "FWCore/Utilities/interface/Exception.h"
+#include "FWCore/Utilities/interface/RandomNumberGenerator.h"
+#include "Geometry/CommonDetUnit/interface/PixelGeomDetUnit.h"
+#include "Geometry/CommonTopologies/interface/PixelTopology.h"
+#include "Geometry/Records/interface/TrackerDigiGeometryRecord.h"
+#include "Geometry/Records/interface/TrackerTopologyRcd.h"
+#include "Geometry/TrackerGeometryBuilder/interface/TrackerGeometry.h"
+
+#include "CLHEP/Random/RandFlat.h"
+#include "CLHEP/Random/RandGauss.h"
+
+/** 
+ * enum to decide which algorithm use to populate the conditions
+ */
+namespace {
+  enum badChannelAlgo { NAIVE = 1, RANDOM = 2, NONE = 99 };
+}
+
+class SiPhase2BadStripChannelBuilder : public ConditionDBWriter<SiStripBadStrip> {
+public:
+  explicit SiPhase2BadStripChannelBuilder(const edm::ParameterSet&);
+  ~SiPhase2BadStripChannelBuilder() override = default;
+
+  static void fillDescriptions(edm::ConfigurationDescriptions& descriptions);
+
+private:
+  std::unique_ptr<SiStripBadStrip> getNewObject() override;
+
+  void algoBeginRun(const edm::Run& run, const edm::EventSetup& es) override {
+    if (!tTopo_) {
+      tTopo_ = std::make_unique<TrackerTopology>(es.getData(topoToken_));
+      tGeom_ = std::make_unique<TrackerGeometry>(es.getData(geomToken_));
+    }
+  };
+
+  void algoAnalyze(const edm::Event& event, const edm::EventSetup& es) override {
+    edm::Service<edm::RandomNumberGenerator> rng;
+    if (!engine) {
+      engine = &rng->getEngine(event.streamID());
+    }
+  }
+
+  std::map<unsigned short, unsigned short> clusterizeBadChannels(
+      const std::vector<Phase2TrackerDigi::PackedDigiType>& maskedChannels);
+
+  // ----------member data ---------------------------
+  std::unique_ptr<TrackerTopology> tTopo_;
+  std::unique_ptr<TrackerGeometry> tGeom_;
+  CLHEP::HepRandomEngine* engine;
```

Comment:
I see here the trailing underscore is used for most cases but for `engine`

Reply:
fixed in the last push.